### DISCUSSION TITLE
Questions about calculating nodal normal for secondary nodes?

### DISCUSSION THREAD

**Discussion by ZhengZQ007** [OpenRadioss/OpenRadioss#1951] [Contributors]
In interface type 24, SUBROUTINE I24NORMNS is used to calculate nodal normal for secondary nodes. For the case of a quadrangle surface, all the nodes of the surface will get contribution from the surface normal. But for the case of a triangle surface, the local nodes 3 and 4 of the surface will get no contribution from the surface normal. Is that reasonable? https://github.com/OpenRadioss/OpenRadioss/blob/7719e93a84957e14b811b106f5c7793fbd8ef874/starter/source/interfaces/inter3d1/i24sti3.F#L1914

**Comment by qiangRadioss** [comment]
Hello,

Radioss uses the same array to store the segment (surface) both for quad&tria, in case of tria, n4=n3.
In I24NORMNS, node from 1-3 are calculated anyway, only n4 is calculated for quad.
where have you seen that node 3 isn't used?

thanks.
Qiang.

**Reply by ZhengZQ007** [reply]
Hi Qiang,

In I24NORMNS, node 3 is calculated by the cross product between line3-4 and line3-2, which is zero for triangle segment because node 3 and node 4 is the same. May node 3 be calculated by the cross product between line3-1 and line3-2 for triangle segment?
![image](https://github.com/OpenRadioss/OpenRadioss/assets/74033106/6126f522-692b-424b-aa7c-b2fbce9e6682)


By the way, In I24FICV_INI, the third fictive node (IP==NPT) normal is calculated by the mean value of the containing segment. But for a triangle segment, node 3 and node 4 seem to be counted twice.
![image](https://github.com/OpenRadioss/OpenRadioss/assets/74033106/dac1321e-4f0f-41bb-b7d2-1fff9f3acb02)
https://github.com/OpenRadioss/OpenRadioss/blob/7719e93a84957e14b811b106f5c7793fbd8ef874/starter/source/interfaces/inter3d1/i24surfi.F#L2087

**Reply by qiangRadioss** [reply]
Hi ZhengZQ007,

yes, you are right, i'll make the fix quickly.

thanks well for this found, appreciate !

Qiang.

**Comment by Schwoertzig** [comment]
It's a good find. thanks.

For the second point, the contribution of N4 is zero if N4=N3. We therefore only count N3 once with a given triangles.

**Comment by qiangRadioss** [comment]
correction done w/ PR#1959
i've done also the correction in I24FICV_INI to avoid potential error.
(the nodal normal is used to avoid wrong initial penetration detection in starter)

Qiang.
